data/txHandler.go

```diff
@@ -34,15 +34,20 @@ import (
 	"github.com/algorand/go-algorand/logging"
 	"github.com/algorand/go-algorand/network"
 	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/util"
 	"github.com/algorand/go-algorand/util/execpool"
 	"github.com/algorand/go-algorand/util/metrics"
 )
 
 // The size txBacklogSize used to determine the size of the backlog that is used to store incoming transaction messages before starting dropping them.
 // It should be configured to be higher then the number of CPU cores, so that the execution pool get saturated, but not too high to avoid lockout of the
 // execution pool for a long duration of time.
-// Set backlog at 'approximately one block' by dividing block size by a typical transaction size.
-var txBacklogSize = config.Consensus[protocol.ConsensusCurrentVersion].MaxTxnBytesPerBlock / 200
+
+// 'approximately one block' by dividing block size by a typical transaction size.
+var txPerBlock = config.Consensus[protocol.ConsensusCurrentVersion].MaxTxnBytesPerBlock / 200
+
+// backlog size is big enough for each peer to have its reserved capacity in the backlog, plus two blocks of shared capacity
+var txBacklogSize = (2 * txPerBlock) + (config.AutogenLocal.IncomingConnectionsLimit * config.AutogenLocal.TxBacklogReservedCapacityPerPeer)
```

Comment:
looks like `AutogenLocal` is not supposed to be used like this
```
// AutogenLocal - this variable is the "input" for the config default generator which automatically updates the above defaultLocal variable.
// it's implemented in ./config/defaults_gen.go, and should be the only "consumer" of this exported variable
```

i.e. `txBacklogSize` should not depend on config or have to be set in MakeTxHandler by using `cfg` as input.

Reply:
Thanks for pointing this out!